Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -770,180 +775,252 @@ section BaseChange
+lemma baseChange_negAddY (x₁ x₂ y₁ L : A) :
+    (W.baseChange B).toAffine.negAddY (φ x₁) (φ x₂) (φ y₁) (φ L) =
+      φ ((W.baseChange A).toAffine.negAddY x₁ x₂ y₁ L) := by
+  erw [← map_negAddY, map_baseChange]
   rfl
 set_option linter.uppercaseLean3 false in
-#align weierstrass_curve.base_change_add_Y'_of_base_change WeierstrassCurve.Affine.baseChange_addY'
+#align weierstrass_curve.base_change_add_Y'_of_base_change WeierstrassCurve.Affine.baseChange_negAddY
 
 lemma baseChange_addY (x₁ x₂ y₁ L : A) :
-    (W.baseChange B).toAffine.addY (ψ x₁) (ψ x₂) (ψ y₁) (ψ L) =
-      ψ ((W.baseChange A).toAffine.addY x₁ x₂ y₁ L) := by
+    (W.baseChange B).toAffine.addY (φ x₁) (φ x₂) (φ y₁) (φ L) =
+      φ ((W.baseChange A).toAffine.addY x₁ x₂ y₁ L) := by
   erw [← map_addY, map_baseChange]
   rfl
 set_option linter.uppercaseLean3 false in
 #align weierstrass_curve.base_change_add_Y_of_base_change WeierstrassCurve.Affine.baseChange_addY
 
 variable {F : Type u} [Field F] [Algebra R F] [Algebra S F] [IsScalarTower R S F]
-  {K : Type v} [Field K] [Algebra R K] [Algebra S K] [IsScalarTower R S K] (ψ : F →ₐ[S] K)
-  {L : Type w} [Field L] [Algebra R L] [Algebra S L] [IsScalarTower R S L] (χ : K →ₐ[S] L)
+  {K : Type v} [Field K] [Algebra R K] [Algebra S K] [IsScalarTower R S K] (φ : F →ₐ[S] K)
+  {L : Type w} [Field L] [Algebra R L] [Algebra S L] [IsScalarTower R S L] (ψ : K →ₐ[S] L)
 
 lemma baseChange_slope (x₁ x₂ y₁ y₂ : F) :
-    (W.baseChange K).toAffine.slope (ψ x₁) (ψ x₂) (ψ y₁) (ψ y₂) =
-      ψ ((W.baseChange F).toAffine.slope x₁ x₂ y₁ y₂) := by
+    (W.baseChange K).toAffine.slope (φ x₁) (φ x₂) (φ y₁) (φ y₂) =
+      φ ((W.baseChange F).toAffine.slope x₁ x₂ y₁ y₂) := by
   erw [← map_slope, map_baseChange]
   rfl
 #align weierstrass_curve.base_change_slope_of_base_change WeierstrassCurve.Affine.baseChange_slope
 
 namespace Point
 
-/-- The function from `W⟮F⟯` to `W⟮K⟯` induced by an algebra homomorphism `ψ : F →ₐ[S] K`,
+/-- The function from `W⟮F⟯` to `W⟮K⟯` induced by an algebra homomorphism `φ : F →ₐ[S] K`,
 where `W` is defined over a subring of a ring `S`, and `F` and `K` are field extensions of `S`. -/
 def mapFun : W⟮F⟯ → W⟮K⟯
   | 0 => 0
-  | some h => some <| (W.baseChange_nonsingular ψ.injective ..).mpr h
+  | some h => some <| (W.baseChange_nonsingular φ.injective ..).mpr h
 #align weierstrass_curve.point.of_base_change_fun WeierstrassCurve.Affine.Point.mapFun
 
-/-- The group homomorphism from `W⟮F⟯` to `W⟮K⟯` induced by an algebra homomorphism `ψ : F →ₐ[S] K`,
+/-- The group homomorphism from `W⟮F⟯` to `W⟮K⟯` induced by an algebra homomorphism `φ : F →ₐ[S] K`,
 where `W` is defined over a subring of a ring `S`, and `F` and `K` are field extensions of `S`. -/
 def map : W⟮F⟯ →+ W⟮K⟯ where
-  toFun := mapFun W ψ
+  toFun := mapFun W φ
   map_zero' := rfl
   map_add' := by
     rintro (_ | @⟨x₁, y₁, _⟩) (_ | @⟨x₂, y₂, _⟩)
     any_goals rfl
     by_cases hx : x₁ = x₂
     · by_cases hy : y₁ = (W.baseChange F).toAffine.negY x₂ y₂
```

Comment:
Note that it's possible to merge two cases in the [definition of add](https://github.com/leanprover-community/mathlib4/commit/022e4c6e420719b0408fe3faa211a06e80e196b0#diff-4acfb4ffccff493de337a21876fa44aed15d232b58d810d3498b488e53c86b2cR729-R733)
and [do one less case split](https://github.com/leanprover-community/mathlib4/commit/022e4c6e420719b0408fe3faa211a06e80e196b0#diff-4acfb4ffccff493de337a21876fa44aed15d232b58d810d3498b488e53c86b2cR956-R963).

Reply:
Thanks.

Reply:
This is far too noisy. I'll change this in a separate PR.